core/src/main/java/org/elasticsearch/cluster/routing/RoutingService.java

```diff
@@ -90,52 +79,26 @@ protected void doStop() {
 
     @Override
     protected void doClose() {
-        FutureUtils.cancel(scheduledRoutingTableFuture);
```

Comment:
not related to this change, but I think it's good to do FutureUtils.cancel(registeredNextDelayFuture) here. Might save on some cryptic error messages during shutdown..